tests/utils/test_post_docs_data.py

```diff
@@ -0,0 +1,49 @@
+import os
+import importlib
+import pytest
+import snowflake.connector
+from utils import post_docs_data
+import responses
+
+class TestPostDocsJob():
+
+    @pytest.mark.parametrize("input_row, output_row",
+                                [
+                                    (('sample feedback 1',''),('sample feedback 1','')),
+                                    (('sample\nfeedback2',''),('sample feedback2','')),
+                                    (('sample\n\nfeedback ""3',''),('sample  feedback 3','')),
+                                    (('sample feedback""""\n""""4',''),('sample feedback 4','')),
+                                    (('sample feedback""""\n""""5','untouched string goes here\n\n""'),('sample feedback 5','untouched string goes here\n\n""'))
+                                ]
+                            )
+    def test_format_row(self, config_feedback, input_row, output_row):
+        # function returns output after removing newline and string quotes
+        assert post_docs_data.format_row(input_row) == output_row
+    
+    def test_feedback_generate_success(self, config_feedback, mock_snowflake_connector, mocker):
+
+        mock_snowflake_connector('utils.post_docs_data')
+        requests_mock = mocker.patch('utils.post_docs_data.requests.post')
+        requests_mock.return_value = mocker.Mock()
+        # Expected to give error since mock object methods cant be called
+        with pytest.raises(ValueError):
+            post_docs_data.post_docs()
```

Comment:
Letting things fail because of incomplete mocking is not a good pattern. A better approach would be to just pass properties as keyword arguments.
```
        requests_mock.return_value = mocker.Mock(status_code=200)
```

However using `responses` seems like a better approach.

Reply:
Removed this test case. Implemented using responses.